src/Cache/MongoStore.php

```diff
@@ -0,0 +1,282 @@
+            ],
+            [
+                [
+                    '$set' => [
+                        'value' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->currentTime()]],
+                                'then' => $this->serialize($value),
+                                'else' => '$value',
+                            ],
+                        ],
+                        'expiration' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->currentTime()]],
+                                'then' => $this->currentTime() + $seconds,
+                                'else' => '$expiration',
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+            ['upsert' => true],
+        );
+
+        return $result->getUpsertedCount() > 0 || $result->getModifiedCount() > 0;
+    }
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param  string $key
+     */
+    #[Override]
+    public function get($key): mixed
+    {
+        $result = $this->collection->findOne(
+            [
+                'key' => $this->prefix . $key,
+            ],
+        );
+
+        if ($result->expiration <= $this->currentTime()) {
+            $this->forgetIfExpired($key);
+
+            return null;
+        }
+
+        return $this->unserialize($result->value);
+    }
+
+    /**
+     * Increment the value of an item in the cache.
+     *
+     * @param  string    $key
+     * @param  int|float $value
+     */
+    #[Override]
+    public function increment($key, $value = 1): int|float|false
+    {
+        $this->forgetIfExpired($key);
+
+        $result = $this->collection->findOneAndUpdate(
+            [
+                'key' => $this->prefix . $key,
+                'expiration' => ['$gte' => $this->currentTime()],
+            ],
+            [
+                '$inc' => ['value' => $value],
+            ],
+            [
+                'returnDocument' => FindOneAndUpdate::RETURN_DOCUMENT_AFTER,
+            ],
+        );
+
+        if (! $result) {
+            return false;
+        }
+
+        if ($result->expiration <= $this->currentTime()) {
+            $this->forgetIfExpired($key);
+
+            return false;
+        }
+
+        return $result->value;
+    }
+
+    /**
+     * Decrement the value of an item in the cache.
+     *
+     * @param  string    $key
+     * @param  int|float $value
+     */
+    #[Override]
+    public function decrement($key, $value = 1): int|float|false
+    {
+        return $this->increment($key, -1 * $value);
+    }
+
+    /**
+     * Store an item in the cache indefinitely.
+     *
+     * @param  string $key
+     * @param  mixed  $value
+     */
+    #[Override]
+    public function forever($key, $value): bool
+    {
+        return $this->put($key, $value, 315360000);
```

Comment:
Should `315360000` be a constant? I'll note that this also isn't an indefinite value. Would it make more sense to have a sentinel value for the expiration timestamp? For instance, zero could be used if you made adjustments to the expiration check (e.g. `forgetIfExpired()`).